site/_includes/masthead.html

```diff
@@ -0,0 +1,21 @@
+{% assign site_logo = site.logo | relative_url %}
+
+<header id="masthead">
+  <div class="inner">
+    <div class="title-area">
+      {% if page.url == '/' %}
+        <h1 class="site-title">
+          <a href="{{ '/' | relative_url }}">
+            {% if site.logo %}<img src="{{ site_logo }}" alt="" class="site-logo">{% endif %}
+          </a>
+        </h1>
+      {% else %}
+        <p class="site-title">
+          <a href="{{ '/' | relative_url }}">
+            {% if site.logo %}<img src="{{ site_logo }}" alt="" class="site-logo">{% endif %}
+          </a>
+        </p>
```

Comment:
there is no text here, but if so, it definitely needs to go inside the link tag. otherwise, it pops below the logo